src/core/blob.cpp

```diff
@@ -11,6 +11,7 @@
 #include "soci/soci-platform.h"
 
 #include <cstddef>
+#include <cassert>
```

Comment:
> I don't think we should ever use assert, it is just evil

I completely disagree with this. Assertions are disabled in release builds, which is what production users will use. Thus, they will not be affected by this.
The way I use assertions is to explicitly encode certain assumptions that a given piece of code makes. This ensures that in case this assumption is broken (which will then correspond to a bug in the implementation or the use of a given piece of code), someone debugging the application (which kinda is the only reason to use non-release builds) will immediately be made aware of this violation. Assertions have the really nice advantage that they report source file and line number whereas exceptions don't do this.

Reply:
No, sorry, I really don't want to go into the details but the fact is that using `assert` is always bad idea (partially because it results in completely different behaviours in debug and release builds) and it's a particularly horrible one for the libraries.

Please don't use it, I can't have using asserts in SOCI on my conscience.

Reply:
> partially because it results in completely different behaviours in debug and release builds

it doesn't. It helps errors are found faster during development. Once your code works, you won't see assertions as they do not depend on runtime conditions (unless you are misusing them).

> assert is always a bad idea 

that's just not true

> Please don't use it, I can't have using asserts in SOCI on my conscience.

For the record, I think this approach is notnjustified, but in the end you are the maintainer, so you make the decisions. I'll remove the asserts.